compiler/rustc_ty_utils/src/abi.rs

```diff
@@ -784,6 +784,21 @@ fn fn_abi_adjust_for_abi<'tcx>(
                 // an LLVM aggregate type for this leads to bad optimizations,
                 // so we pick an appropriately sized integer type instead.
                 arg.cast_to(Reg { kind: RegKind::Integer, size });
+
+                // Let's see if we can add a `noundef`. This is only legal for arrays, definitely
```

Comment:
I *think* the elephant in the room here is padding, isn't it? Arrays are fine as they do not have any padding. But the word "padding" does not appear anywhere in the PR diff or description, so I am confused.